### PR TITLE
17506 - Magic Link Affiliation cleanup for Corp and Co-ops

### PR DESCRIPTION
**Pull request by rodrigo-barraza**
*Issue #:17506*
https://github.com/bcgov/entity/issues/17506

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/manage-business/manage-business-dialog/ManageBusinessDialog.vue [hunk 1]
```diff
@@ -173,15 +173,15 @@
                 </v-list-group>
 
                 <v-list-group
-                  v-if="(isBusinessLegalTypeCorporation || isBusinessLegalTypeCoOp || isBusinessLegalTypeFirm) && businessContactEmail"
+                  v-if="(isBusinessLegalTypeCorporation || isBusinessLegalTypeBenefit || isBusinessLegalTypeCoOp || isBusinessLegalTypeFirm) && businessContactEmail"
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/manage-business-dialog/ManageBusinessDialog.vue:176]: Getting long here, I'd almost recommend putting into a computed

**Review by seeker25 — APPROVED** [review]
Would recommend refactoring some of the refs into reactive, and using 
```
return ...toRefs(<reactive>) 
```

It would get rid of all of the .value's

**Comment by seeker25** [comment]
`Error:   557:10  error  'CreateAffiliationInvitation' is defined but never used  @typescript-eslint/no-unused-vars`

File: auth-web/src/components/auth/manage-business/manage-business-dialog/ManageBusinessDialog.vue [hunk 2]
```diff
@@ -476,7 +480,7 @@ export default defineComponent({
       const hasPasscode = !!passcode.value
       const hasCertified = !!isCertified.value
       const isCertifiedBy = !!certifiedBy.value
-      if (isBusinessLegalTypeCorporation.value || isBusinessLegalTypeCoOp.value) {
+      if (isBusinessLegalTypeCorporation.value || isBusinessLegalTypeBenefit || isBusinessLegalTypeCoOp.value) {
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/manage-business-dialog/ManageBusinessDialog.vue:483]: could put these into a function?

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2505-wcnjhx4s.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2505-wcnjhx4s.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2505-wcnjhx4s.web.app

**Comment by seeker25** [comment]
@hfekete added some small changes to removal / markup

File: auth-web/.eslintrc.js [hunk 3]
```diff
@@ -35,6 +35,7 @@ module.exports = {
     'vue/no-v-for-template-key-on-child': 'warn',
     'vue/no-v-html': 'off',
     'vue/v-on-event-hyphenation': 'off',
+    'vue/no-ref-as-operand': 'error',
```
Comment (seeker25) [inline auth-web/.eslintrc.js:38]: Should stop accidently not including `.value`  etc.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue [hunk 4]
```diff
@@ -286,6 +286,8 @@ export default defineComponent({
             statusText = '<strong>Not Authorized</strong>. Your request to manage this business has been declined.'
             break
           case AffiliationInvitationStatus.Expired:
+            statusText = 'Invitation is <strong>Expired</strong>.'
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:289]: @hfekete added this in, will have to check with Ethan if it's correct - I don't think yours expires?

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue [hunk 5]
```diff
@@ -49,23 +49,7 @@
...
-                  <span class="pl-1">Delete {{ tempDescription(item) }}</span>
-                </v-list-item-subtitle>
-                <v-list-item-subtitle v-else>
-                  <v-icon small>mdi-delete</v-icon>
-                  <span class="pl-1">Remove From Table</span>
-                </v-list-item-subtitle>
-              </v-list-item>
-              <v-list-item
-                v-else
+                v-if="item.affiliationInvites[0].status !== 'ACCEPTED'"
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/AffiliationAction.vue:52]: @hfekete changed the logic a tad

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue [hunk 6]
```diff
@@ -280,7 +265,17 @@ export default defineComponent({
     }
 
     /** Emit business/nr information to be unaffiliated. */
-    const removeBusiness = (business: Business): void => {
+    const removeBusiness = async (business: Business): Promise<void> => {
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/AffiliationAction.vue:268]: Should remove affiliation invitations as well provided they're in PENDING, FAILED, EXPIRED.

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue [hunk 7]
```diff
@@ -335,6 +330,7 @@ export default defineComponent({
 
     // Actions
     const actionHandler = async (business: Business) => {
+      debugger
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/AffiliationAction.vue:333]: yikes will fix

File: auth-web/src/components/auth/manage-business/AffiliationAction.vue [hunk 8]
```diff
@@ -279,8 +264,18 @@ export default defineComponent({
       }
     }
 
-    /** Emit business/nr information to be unaffiliated. */
-    const removeBusiness = (business: Business): void => {
+    /** Remove business/nr affiliation or affiliation invitation. */
+    const removeAffiliationOrInvitation = async (business: Business): Promise<void> => {
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/AffiliationAction.vue:268]: @hfekete just so you're aware

**Pull request merged**